pallets/claims/src/lib.rs

```diff
@@ -0,0 +1,207 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+use codec::{Decode, Encode};
+use frame_support::{
+	decl_error, decl_event, decl_module, decl_storage,
+	dispatch::DispatchResult,
+	ensure,
+	traits::{Currency, Get, Imbalance},
+	weights::{DispatchClass, Pays},
+};
+use frame_system::ensure_signed;
+use primitives::Balance;
+use sp_runtime::traits::Zero;
+use sp_std::prelude::*;
+use sp_std::vec::Vec;
+
+use frame_support::sp_runtime::traits::{DispatchInfoOf, SignedExtension};
+use frame_support::sp_runtime::transaction_validity::{
+	InvalidTransaction, TransactionValidity, TransactionValidityError, ValidTransaction,
+};
+use frame_support::traits::IsSubType;
+use sp_std::marker::PhantomData;
+
+pub use traits::*;
+
+mod claims_data;
+mod migration;
+mod traits;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+mod benchmarking;
+
+use weights::WeightInfo;
+
+pub mod weights;
+
+pub trait Config: frame_system::Config {
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+	type Prefix: Get<&'static [u8]>;
```

Comment:
is there any reason for the prefix being configurable?

Reply:
yes, it was designed bearing in mind we might want to publish the claims as a standalone pallet

Reply:
It can be used by anybody else then as it's pretty generic